### PR TITLE
ci: add windows test support

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
This change adds support for windows test, in its basic form, for codebuild.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild_specs/run_e2e_tests_windows.yml [hunk 1]
```diff
@@ -25,9 +25,7 @@ env:
 phases:
   build:
     commands:
-      - export NODE_OPTIONS=--max-old-space-size=8096
-      - export AMPLIFY_DIR=$CODEBUILD_SRC_DIR/out
-      - export AMPLIFY_PATH=$CODEBUILD_SRC_DIR/out/amplify-pkg-linux-x64
-      - echo $AMPLIFY_DIR
-      - echo $AMPLIFY_PATH
-      - source ./shared-scripts.sh && _runE2ETestsWindows
+      - choco install -fy jq
```
Comment (awsluja) [inline codebuild_specs/run_e2e_tests_windows.yml:28]: I will be adding this to the base image at a later date, it only takes a minute to install this dependency.

File: codebuild_specs/scripts-windows/shared-scripts-windows.sh [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    s3Path="s3://$CACHE_BUCKET_NAME/$CODEBUILD_SOURCE_VERSION/$alias"
+    echo writing cache to $s3Path
+    # zip contents and upload to s3
+    # if ! (cd $localPath && tar -czf cache.tar . && ls && aws s3 cp cache.tar $s3Path); then
+    #     echo Something went wrong storing the cache.
+    # fi
+    if ! (cd $localPath && tar -czf cache.tar . && ls && aws s3 cp cache.tar $s3Path); then
+        echo Something went wrong storing the cache.
+    fi
+    echo done writing cache
```
Comment (evcodes) [inline codebuild_specs/scripts-windows/shared-scripts-windows.sh:26]: Will this work?

**Review by sdstolworthy — APPROVED** [review]

**Review by goldbez — DISMISSED** [review]

File: codebuild_specs/scripts-windows/shared-scripts-windows.sh (hunk 2 above)
Comment (sdstolworthy) [reply codebuild_specs/scripts-windows/shared-scripts-windows.sh:26]: The echo will work, the syntax highlighter is just being a little overzealous.

Comment (awsluja) [reply codebuild_specs/scripts-windows/shared-scripts-windows.sh:26]: yeah it works

**Review by sdstolworthy — APPROVED** [review]

**Review by goldbez — APPROVED** [review]

**Pull request merged**